src/routers/courseItem.ts

```diff
@@ -66,15 +68,17 @@ cItemRouter.put('/', async (req, res) => {
 		const userID = req.header('userID');
 		if (userID) {
 			const body = req.body;
+			console.log('updating course item with id ' + body.id);
+			const name = body.name;
+			const weight = numberify(body.weight);
+			const grade = numberify(body.grade);
+			const date = body.date;
 			const courseItem = new CourseItem({
 				id: body.id,
-				owner: userID,
-				name: body.name,
-				weight: numberify(body.weight),
-				grade: numberify(body.grade),
-				date: body.date,
-				createdAt: body.createdAt,
-				updatedAt: body.updatedAt,
+				...(name ? { name } : {}),
```

Comment:
If someone doesn't provide a name then updates, does the name get removed from the database or is it just considered a field not to update? I am asking because some APIs for databases consider an omitted key as don't change. If that is the case, just add the fields always unconditionally

Reply:
Hey! This actually had to be made like this because an empty key removes the field from the API, instead of ignoring it (this way, if a field is empty the key isn't included in the object at all). Let me know if this makes sense or if there are any better ways of going about this!